File: src/TaskOTime/TaskOTime.TimeTrackingServices/IActionTarget.vb
Namespace ActiveDevelop.TimeTrackingServices
    Public Interface IActionTarget(Of ActionTargetIDType As {Structure, IComparable(Of ActionTargetIDType)})
        Inherits IComparable(Of IActionTarget(Of ActionTargetIDType))

        Property Id As ActionTargetIDType
    End Interface
End Namespace

File: src/TaskOTime/TaskOTime.TimeTrackingServices/IActionTarget.cs
using System;

namespace ActiveDevelop.TimeTrackingServices
{
    public interface IActionTarget<ActionTargetIDType> : IComparable<IActionTarget<ActionTargetIDType>> where ActionTargetIDType : struct, IComparable<ActionTargetIDType>
    {

        ActionTargetIDType Id { get; set; }
    }
}